frontend/src/views/CalculateNutrients/CalculateNutrientsComponents/FertilizerModal.tsx

```diff
@@ -179,29 +240,98 @@ export default function FertilizerModal({
         setFertilizerUnits(data);
       }
     });
+
+    apiCache.callEndpoint('api/densityunits/').then((response) => {
+      if (response.status === 200) {
+        const { data } = response;
+        setDensityUnits(data);
+      }
+    });
+
+    apiCache.callEndpoint('api/liquidfertilizerdensities/').then((response) => {
+      if (response.status === 200) {
+        const { data } = response;
+        setLiqDensityFactors(data);
+      }
+    });
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, []);
 
-  const handleModalCalculate = (e: FormEvent) => {
-    e.preventDefault();
+  const isCustomFertilizerValid = () => {
+    if (formCustomFertilizer.nitrogen || formCustomFertilizer.nitrogen === 0) return false;
+    if (formCustomFertilizer.nitrogen || formCustomFertilizer.phosphorous === 0) return false;
+    if (formCustomFertilizer.nitrogen || formCustomFertilizer.potassium === 0) return false;
+
+    return true;
+  };
+
+  const isRegularFertilizerValid = () => {
+    if (formState.fertilizerId) return true;
```

Comment:
You could also write `return Boolean(formState.fertilizerId)` or `return !!formState.fertilizerId`

Reply:
Simplified.